### PR TITLE
fix(bigquery): address possible panic due to offset checking in handleInsertErrors 

### PR DESCRIPTION
**Pull request by shollyman**
External reporter identified case where improper bounds checking can
trigger panic when comparing the index value from structured error
response.

Fixes: #3519

### CODE REVIEW

File: bigquery/inserter_test.go [hunk 1]
```diff
@@ -138,10 +144,29 @@ func TestHandleInsertErrors(t *testing.T) {
...
+				{Errors: []*bq.ErrorProto{{Message: "m0"}}, Index: 2},
+			},
+			want: fmt.Errorf("internal error: unexpected row index: 2"),
+		},
 	} {
 		got := handleInsertErrors(test.in, rows)
-		if !testutil.Equal(got, test.want) {
-			t.Errorf("%#v:\ngot\n%#v\nwant\n%#v", test.in, got, test.want)
+		_, ok := got.(PutMultiError)
+		if ok {
```
Comment (tbpg) [inline bigquery/inserter_test.go:157]:
```suggestion
		if _, ok := got.(PutMultiError); ok {
```

File: bigquery/inserter_test.go [hunk 2]
```diff
@@ -138,10 +144,29 @@ func TestHandleInsertErrors(t *testing.T) {
...
+			// compare structure of the PutMultiError
+			if !testutil.Equal(got, test.want) {
+				t.Errorf("(case: %s)\nin %#v\ngot\n%#v\nwant\n%#v", test.description, test.in, got, test.want)
+			}
+		} else {
```
Comment (tbpg) [inline bigquery/inserter_test.go:162]:
```suggestion
			}
			continue
		}
```

Comment (shollyman) [reply bigquery/inserter_test.go:161]: Unsure I'm following.  We still want to probe errors when not a PutMultiError, not simply move to the next case.

Comment (shollyman) [reply bigquery/inserter_test.go]: Revisited.  Continue inside the type assertion case, and then do the secondary check.

**Review by codyoss — APPROVED** [review]
LGTM

**Pull request merged**